assets/js/Viewer.jsx

```diff
@@ -18,16 +18,28 @@ class Viewer extends Component {
       <div>
         {this.sortedKeys().map((key) => {
           const { signs } = this.props;
+          const { currentTime } = this.props;
           const lines = signs[key];
-          return <Sign key={key} signId={key} lineOne={lines[0]} lineTwo={lines[1]} />;
+          return (
+            <Sign
+              key={key}
+              signId={key}
+              lineOne={lines[0].text}
+              lineOneDuration={lines[0].duration}
+              lineTwo={lines[1].text}
+              lineTwoDuration={lines[1].duration}
+              currentTime={currentTime}
+            />
+          );
         })}
       </div>
     );
   }
 }
 
 Viewer.propTypes = {
-  signs: PropTypes.objectOf(PropTypes.arrayOf(PropTypes.string)).isRequired,
+  signs: PropTypes.objectOf(PropTypes.arrayOf(PropTypes.object)).isRequired,
```

Comment:
I'd use `PropTypes.shape({ attr: PropTypes.string, ... })` here instead of just PropTypes.object

Reply:
oh, thats good to know about, thanks!